src/server/scripts/Northrend/Ulduar/Ulduar/boss_freya.cpp

```diff
@@ -360,7 +337,7 @@ class boss_freya : public CreatureScript
                         continue;
 
                     if (Creature* e = ObjectAccessor::GetCreature(*me, _elderGUID[i]))
-                        Unit::Kill(e, e);
+                        e->AI()->DoAction(ACTION_ELDER_FREYA_KILLED);
```

Comment:
```cpp
e->DespawnOrUnsummon();
```

Reply:
@Nyeriah Sorry, didn't know. And `e->DespawnOrUnsummon();`  is better solution,